test/common/upstream/health_check_fuzz.cc

```diff
@@ -112,34 +107,161 @@ void HealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+    expectClientCreate();
+    interval_timer_->invokeCallback();
+  }
+}
+
+void TcpHealthCheckFuzz::triggerIntervalTimer() {
+  if (!interval_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Interval timer is disabled. Skipping trigger interval timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered interval timer");
+  interval_timer_->invokeCallback();
+}
+
+void TcpHealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+  if (!timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping trigger timeout timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered timeout timer");
+  timeout_timer_->invokeCallback(); // This closes the client, turns off timeout
+                                    // and enables interval
+  if (!last_action) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from network timeout");
+    expectClientCreate();
+    interval_timer_->invokeCallback();
+  }
+}
+
+void TcpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type, bool last_action) {
+  // On a close event, the health checker will call handleFailure if expect_close_ is false. This is
+  // set by multiple code paths. handleFailure() turns on interval and turns off timeout. However,
+  // other action of the fuzzer account for this by explicility invoking a client after
+  // expect_close_ gets set to true, turning expect_close_ back to false.
+  connection_->raiseEvent(event_type);
+  if (!last_action && event_type != Network::ConnectionEvent::Connected) {
+    if (!interval_timer_->enabled_) {
+      return;
+    }
+    ENVOY_LOG_MISC(trace, "Creating client from close event");
+    expectClientCreate();
+    interval_timer_->invokeCallback();
+  }
+
+  // In the specific case of:
+  // https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/health_checker_impl.cc#L489
+  // This blows away client, should create a new one
+  if (event_type == Network::ConnectionEvent::Connected && empty_response_) {
+    ENVOY_LOG_MISC(trace, "Creating client from connected event and empty response.");
+    expectClientCreate();
+    interval_timer_->invokeCallback();
+  }
+}
+
+Network::ConnectionEvent
+HealthCheckFuzz::getEventTypeFromProto(const test::common::upstream::RaiseEvent& event) {
   switch (event) {
   case test::common::upstream::RaiseEvent::CONNECTED: {
-    eventType = Network::ConnectionEvent::Connected;
+    return Network::ConnectionEvent::Connected;
     break;
   }
   case test::common::upstream::RaiseEvent::REMOTE_CLOSE: {
-    eventType = Network::ConnectionEvent::RemoteClose;
+    return Network::ConnectionEvent::RemoteClose;
     break;
   }
   case test::common::upstream::RaiseEvent::LOCAL_CLOSE: {
-    eventType = Network::ConnectionEvent::LocalClose;
+    return Network::ConnectionEvent::LocalClose;
     break;
   }
   default: // shouldn't hit
-    eventType = Network::ConnectionEvent::Connected;
+    return Network::ConnectionEvent::Connected;
```

Comment:
You could add a NOT_REACHED_GCOVR_EXCL_LINE line here, which is usually typical of how Envoy deals with unreachable codepaths